pkgs/development/haskell-modules/configuration-common.nix

```
@@ -2372,6 +2372,27 @@ self: super: {
   # https://github.com/kuribas/mfsolve/issues/8
   mfsolve = dontCheck super.mfsolve;
 
+  knob = appendPatch ./patches/knob-ghc9.patch super.knob;
+
+  # compatibility with random-fu 0.3 https://github.com/mokus0/misfortune/pull/5
+  misfortune = appendPatch ./patches/misfortune-ghc9.patch (overrideCabal (drv: {
+    revision = "0";
+    editedCabalFile = "sha256-17uKoAt7bCG3XB2iwi8udOyuci+go/gH1XbxmM3dPoc=";
+  }) super.misfortune);
+
+  # GHC 9 support https://github.com/lambdabot/dice/pull/2
+  dice = appendPatch ./patches/dice-ghc9.patch (overrideCabal (drv: {
+    revision = "0";
+    editedCabalFile = "sha256-Ydw9YiNZ8V0naHjfYhE1m5P7M45wZQu4iTZ5ycf4X4c=";
```

To get the original cabal file.

```suggestion
    revision = null;
    editedCabalFile = null;
```